tests/integration/test_sns.py

```diff
@@ -2179,9 +2199,57 @@ def test_subscribe_external_http_endpoint(
         subscribe_url = payload["SubscribeURL"]
         service_url, subscribe_url_path = payload["SubscribeURL"].rsplit("/", maxsplit=1)
         assert subscribe_url == (
-            f"{service_url}/?Action=ConfirmSubscription" f"&TopicArn={topic_arn}&Token={token}"
+            f"{service_url}/?Action=ConfirmSubscription&TopicArn={topic_arn}&Token={token}"
+        )
+
+        test_broken_confirm_url = (
+            f"{service_url}/?Action=ConfirmSubscription&TopicArn=not-an-arn&Token={token}"
+        )
+        broken_confirm_subscribe_request = requests.get(test_broken_confirm_url)
+        snapshot.match(
+            "broken-topic-arn-confirm",
+            _get_snapshot_requests_response(broken_confirm_subscribe_request),
+        )
+
+        test_broken_token_confirm_url = (
+            f"{service_url}/?Action=ConfirmSubscription&TopicArn={topic_arn}&Token=abc123"
+        )
+        broken_token_confirm_subscribe_request = requests.get(test_broken_token_confirm_url)
+        snapshot.match(
+            "broken-token-confirm",
+            _get_snapshot_requests_response(broken_token_confirm_subscribe_request),
         )
 
+        # a topic with a different region will fail than the subscription
```

Comment:
nit; I am not quite sure what this means, please clarify :) (possibly typo, did you mean *will then fail the subscription*?)